configs/nvme-storage-configurator/configmap.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nvme-storage-configurator
+  namespace: nvme-storage-config
+data:
+  configure.sh: |
+    #!/usr/bin/env bash
+
+    # Enable debugging
+    set -x
+
+    cp -f /scripts/configuration.sh /host-usr-local-bin/configuration.sh
+    chmod 0755 /host-usr-local-bin/configuration.sh
+    
+    cp -f /scripts/nvme-storage.service /host-etc-systemd-dir/nvme-storage.service
+    chmod 0644 /host-etc-systemd-dir/nvme-storage.service
+
+    nsenter -t 1 -m -u -i -n -p -- systemctl daemon-reload
+    nsenter -t 1 -m -u -i -n -p -- systemctl enable --now nvme-storage.service
+    nsenter -t 1 -m -u -i -n -p -- sleep 10
+    nsenter -t 1 -m -u -i -n -p -- restorecon -R -v -F /home/nvme
+    nsenter -t 1 -m -u -i -n -p -- restorecon -R -v -F /var/lib/containers
+
+    if [ $? -eq 0 ]; then
+        echo "The last command was successful. Exiting gracefully."
+        exit 0
+    else
+        echo "The last command failed. Exiting with an error."
+        exit 1
+    fi
+
+  configuration.sh: |
+    #!/usr/bin/env bash
+
+    set -x
+
+    export DEVICE="/dev/nvme1n1"
+    export MOUNT_POINT="/home/nvme"
+    export CONTAINER_DIR="/var/lib/containers"
+    export FSTYPE="xfs"
+
+    mkdir -p ${MOUNT_POINT}
+
+    if lsblk -no FSTYPE "$DEVICE" | grep -qE '\S'; then
+        echo "File system already exists on $DEVICE."
+    else
+        echo "No file system found on $DEVICE. Creating XFS filesystem..."
+        mkfs -t "$FSTYPE" "$DEVICE"
+    fi
+
+    if ! grep -q "$DEVICE $MOUNT_POINT" /etc/fstab; then
+        echo "$DEVICE $MOUNT_POINT $FSTYPE defaults 0 0" >> /etc/fstab
+    fi
+
+    mount -a
+
+    mount --bind ${CONTAINER_DIR} ${MOUNT_POINT}
+
+    echo "Filesystem setup and mounting complete."
+
+  nvme-storage.service: |
+    [Unit]
+    Description=Custom Service for NVMe Storage
+    DefaultDependencies=no
+    Before=kubelet.service network-pre.target
```

Comment:
I think there is a conflict here. Since this script is called using a daemonset, the unit can't start before the kubelet.

Reply:
@gbenhaim for the first time, DS is starting the service manually. After the node reboots, this service will run before kublet service. If it's fundmentally incorrect, let's discuss this.